includes/multi-currency/Compatibility.php

```diff
@@ -62,15 +72,18 @@ public function should_hide_widgets(): bool {
 	 */
 	public function should_convert_product_price( $product = null ): bool {
 		// If we have a product, and it's a subscription renewal.
-		if ( $product && $this->is_product_subscription_renewal( $product ) ) {
-			$calls = [
-				'WC_Cart_Totals->calculate_item_totals',
-				'WC_Cart->get_product_subtotal',
-				'wc_get_price_excluding_tax',
-				'wc_get_price_including_tax',
-			];
-			if ( $this->utils->is_call_in_backtrace( $calls ) ) {
-				return false;
+		if ( $product ) {
+			if ( $this->is_product_subscription_type_in_cart( $product, 'renewal' )
+				|| $this->is_product_subscription_type_in_cart( $product, 'resubscribe' ) ) {
+				$calls = [
+					'WC_Cart_Totals->calculate_item_totals',
+					'WC_Cart->get_product_subtotal',
+					'wc_get_price_excluding_tax',
+					'wc_get_price_including_tax',
+				];
+				if ( $this->utils->is_call_in_backtrace( $calls ) ) {
+					return false;
+				}
```

Comment:
What do you think of reducing the indentation level here by inverting the ifs? Would it make this easier to read? I haven't tried this, and it might not make much sense here, so I'll defer it to you 😄

Reply:
Inverting the if for the product and adding an early return does make it easier to read: 

```
	public function should_convert_product_price( $product = null ): bool {
		if ( ! $product ) {
			return true;
		}

		// Check for subscription renewal or resubscribe.
		if ( $this->is_product_subscription_type_in_cart( $product, 'renewal' )
			|| $this->is_product_subscription_type_in_cart( $product, 'resubscribe' ) ) {
			$calls = [
				'WC_Cart_Totals->calculate_item_totals',
				'WC_Cart->get_product_subtotal',
				'wc_get_price_excluding_tax',
				'wc_get_price_including_tax',
			];
			if ( $this->utils->is_call_in_backtrace( $calls ) ) {
				return false;
			}
		}

		return true;
	}
```

I _think_ further nesting could be removed by changing:

```
		if ( $this->is_product_subscription_type_in_cart( $product, 'renewal' )
			|| $this->is_product_subscription_type_in_cart( $product, 'resubscribe' ) ) {
				...
```

to this

```
		if ( ! $this->is_product_subscription_type_in_cart( $product, 'renewal' )
			&& ! $this->is_product_subscription_type_in_cart( $product, 'resubscribe' ) ) {
				return true;
```

Is that what you were thinking?

Reply:
Just noting, I updated to the first chunk here in https://github.com/Automattic/woocommerce-payments/pull/2367/commits/8d55540622e6a1b5f52c9ff1b2d876e2fa137fc3

Reply:
Awesome! I like how it looks like after removing the `$product` check nesting 🔥. If you find that the second change is not that beneficial we could leave it like it is right now and merge this PR. 🚀